### PR TITLE
Fix entity category of Home Assistant exposed enum sensors

### PR DESCRIPTION
**Pull request by frenck**
Similar #19434, but in this case, for the enums that are exposed as sensors.

Related to #19430

### CODE REVIEW

**Comment by hmmbob** [comment]
My errors all have `power_outage_memory` as suffix, those aren't touched in the pr. Is that expected?

**Comment by frenck** [comment]
@hmmbob, yes, it will be corrected by: 

https://github.com/Koenkk/zigbee2mqtt/pull/19474/files#diff-c63411dead8da381e5b5e1577175477e6536b9c8d4ad997cd701a3f10d46d233R908-R912

**Comment by hmmbob** [comment]
Right, I see. Thanks, I'll rest as soon as it's in dev

File: lib/extension/homeassistant.ts [hunk 1]
```diff
@@ -903,7 +903,15 @@ export default class HomeAssistant extends Extension {
                         enabled_by_default: !(firstExpose.access & ACCESS_SET),
                         ...lookup[firstExpose.name],
                     },
-                });
+                };
+
+                // If it has an entity category of config, but exposed as sensor, then change
+                // it to diagnostic. Sensors have no input, so can't be configured.
+                if (discoveryEntry.discovery_payload.entity_category === 'config') {
```
Comment (Koenkk) [inline lib/extension/homeassistant.ts:910]: @frenck if I understand correctly the logic is: only if `type` == `sensor` and `entity_category` == `config`, then `entity_category` should be set to `diagnostic`. If true, then I propose to remove this and the other similar if and instead change the end of the function to:

```js
        discoveryEntries.forEach((d) => {
            // If a sensor has entity category `config`, then change
            // it to `diagnostic`. Sensors have no input, so can't be configured.
            // https://github.com/Koenkk/zigbee2mqtt/pull/19474
            if (d.type === 'sensor' && d.discovery_payload.entity_category === 'config') {
                d.discovery_payload.entity_category === 'diagnostic';
            }
        });

        return discoveryEntries;
```

**Comment by frenck** [comment]
Adjusted as suggested 👍

**Comment by Koenkk** [comment]
Thanks as always 😄

File: lib/extension/homeassistant.ts [hunk 2]
```diff
@@ -971,6 +965,15 @@ export default class HomeAssistant extends Extension {
             throw new Error(`Unsupported exposes type: '${firstExpose.type}'`);
         }
 
+        discoveryEntries.forEach((d) => {
+            // If a sensor has entity category `config`, then change
+            // it to `diagnostic`. Sensors have no input, so can't be configured.
+            // https://github.com/Koenkk/zigbee2mqtt/pull/19474
+            if (d.type === 'sensor' && d.discovery_payload.entity_category === 'config') {
```
Comment (elupus) [inline lib/extension/homeassistant.ts:972]: This seem to have missed binary sensors, which now also are validated.

Comment (Koenkk) [reply lib/extension/homeassistant.ts:972]: Pushed a fix!

Comment (frenck) [reply lib/extension/homeassistant.ts:972]: > This seem to have missed binary sensors, which now also are validated.

This statement is not correct. This has not been implemented for binary sensors (but likely would in the future).

Comment (elupus) [reply lib/extension/homeassistant.ts:972]: It was implemented for mqtt integration.

Comment (frenck) [reply lib/extension/homeassistant.ts:972]: Huh? Why would we implement such rules on an integration level? That makes no sense...

Will investigate that one.

Comment (frenck) [reply lib/extension/homeassistant.ts:972]: Ok, so this sneaked in for 2023.11.1 unintentionally, specifically on MQTT.
We are going to revert that part and shop 2023.11.2.

The change by Koen is still good, as that is expected behavior for 2023.12 (which got merged today as well).

../Frenck

**Pull request merged**